platform/macos/godot_application_delegate.h

```diff
@@ -41,6 +41,7 @@
 - (void)forceUnbundledWindowActivationHackStep2;
 - (void)forceUnbundledWindowActivationHackStep3;
 - (void)handleAppleEvent:(NSAppleEventDescriptor *)event withReplyEvent:(NSAppleEventDescriptor *)replyEvent;
+- (void)handleAppleEvent:(NSAppleEventDescriptor *)event onAppUrlEvent:(NSAppleEventDescriptor *)replyEvent;
```

Comment:
It will work as is, but maybe better to rename these function to something more readable instead of `handleAppleEvent` with a different argument name, like:

```suggestion
- (void)handleFileOpenEvent:(NSAppleEventDescriptor *)event withReplyEvent:(NSAppleEventDescriptor *)replyEvent;
- (void)handleURLOpenEvent:(NSAppleEventDescriptor *)event withReplyEvent:(NSAppleEventDescriptor *)replyEvent;
```

Reply:
I think we should actually name the objective-c functions properly as naming them withReplyEvent to me doesn't make sense.

I think the correct way would be to do the apple guidelines for this name
- onAppURLEvent
- onAppOpenDocument

Otherwise to me withReplyEvent is not descriptive as-is

Reply:
Well, Apple usually do name methods by a first keyword and the rest is `withSomething`.

But, actually `replyEvent` is not used at all, so can be dropped at all and reduced to :

```c++
 (void)onAppOpenDocument:(NSAppleEventDescriptor *)event;
 (void)onAppURLEvent:(NSAppleEventDescriptor *)event;
```

Reply:
Sounds good, will refactor to this!